packages/middleware/__tests__/integration/createServer.spec.ts

```diff
@@ -141,4 +153,35 @@ describe("[Integration] Create server", () => {
       },
     });
   });
+
+  it("should make a call to a namespaced method", async () => {
+    expect.assertions(2);
+
+    const { status, text } = await request(app)
+      .post("/test_integration/my-namespaced-extension/myFuncNamespaced")
+      .send([]);
+
+    const response = JSON.parse(text);
+    // This is the result of the original "success" function from the integration
+    const apiMethodResult = await success();
+
+    expect(status).toEqual(200);
+    expect(response).toEqual(apiMethodResult);
+  });
+
+  it("namespaced extension should be not merged to the shared api", async () => {
+    expect.assertions(2);
+
+    const { status, text } = await request(app)
+      .post("/test_integration/myFunc")
+      .send([]);
+
+    const response = JSON.parse(text);
+    // This is the result of the original "success" function from the integration
+    const apiMethodResult = await success();
+
+    // If merged, the response would be { message: "error", error: true, status: 404 }
+    expect(status).toEqual(200);
+    expect(response).toEqual(apiMethodResult);
+  });
```

Comment:
Let's add 2 tests:
- calling not existing endpoint in extension's namespace, e.g. `post("/fruits/dog")`,
- calling endpoint in not existing extension's namespace, e.g. `post("/jhasdjhgb/xvewdeq")`.

It should be catched in try...catch block and passed to errorHandler. Let's make sure default error handler won't throw error in that case

Reply:
Ok, I added more tests and rework function resolving logic.